lib/action_controller/serialization.rb

```diff
@@ -25,6 +25,10 @@ def get_serializer(resource, options = {})
           "Please pass 'adapter: false' or see ActiveSupport::SerializableResource#serialize"
         options[:adapter] = false
       end
+      if options[:pagination]
+        options[:original_url] = original_url
+        options[:query_parameters] = query_parameters
+      end
```

Comment:
> ```ruby
options[:pagination][:original_url] ||= original_url
options[:pagination][:query_parameters] ||= query_parameters
```

:+1: 

I'm also in favor or building the self link based on this.
It seems to me that some of the changes proposed here are implemented already, right @bacarini?